src/frontend/src/binder/expr/mod.rs

```diff
@@ -278,22 +278,36 @@ impl Binder {
                 false => non_const_exprs.push(expr),
             }
         }
-        let mut ret = FunctionCall::new(ExprType::In, bound_expr_list)?.into();
+
+        let mut ret = if bound_expr_list.len() == 1 {
+            None
+        } else {
+            Some(FunctionCall::new(ExprType::In, bound_expr_list)?.into())
+        };
         // Non-const exprs are not part of IN-expr in backend and rewritten into OR-Equal-exprs.
         for expr in non_const_exprs {
-            ret = FunctionCall::new(
-                ExprType::Or,
-                vec![
-                    ret,
-                    FunctionCall::new(ExprType::Equal, vec![left.clone(), expr])?.into(),
-                ],
-            )?
-            .into();
+            if ret.is_none() {
+                ret = Some(FunctionCall::new(ExprType::Equal, vec![left.clone(), expr])?.into());
+            } else {
+                ret = Some(
+                    FunctionCall::new(
+                        ExprType::Or,
+                        vec![
+                            ret.unwrap(),
```

Comment:
nit: `if-let` or `let-else` to avoid this `unwrap()`

Reply:
Adding 2 test cases here.